### PR TITLE
tracing: plumb HttpTracer into AsyncClientImpl

### PR DESCRIPTION
**Pull request by goaway**

### CODE REVIEW

File: source/common/router/router.h [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #include "envoy/upstream/cluster_manager.h"
 
 #include "common/common/logger.h"
+#include "common/tracing/http_tracer_impl.h"
```
Comment (mattklein123) [inline source/common/router/router.h]: shouldn't need impl header here, just interface.

File: include/envoy/tracing/http_tracer.h [hunk 2]
```diff
@@ -67,6 +67,8 @@ class HttpTracer {
 public:
   virtual ~HttpTracer() {}
 
+  virtual void installDriver(DriverPtr&& driver) PURE;
```
Comment (mattklein123) [inline include/envoy/tracing/http_tracer.h]: I see why you need this in the config code, but I would prefer that we not expose this function around the code base. If you need it to be part of the interface, can we make another interface the derives from this interface that allows driver installation?

File: source/common/tracing/http_tracer_impl.cc [hunk 3]
```diff
@@ -158,6 +158,8 @@ void HttpTracerUtility::finalizeSpan(Span& active_span, const Http::HeaderMap& r
 HttpTracerImpl::HttpTracerImpl(DriverPtr&& driver, const LocalInfo::LocalInfo& local_info)
     : driver_(std::move(driver)), local_info_(local_info) {}
 
+void HttpTracerImpl::installDriver(DriverPtr&& driver) { driver_ = std::move(driver); }
```
Comment (mattklein123) [inline source/common/tracing/http_tracer_impl.cc]: No one should start a span between the time we make a tracer, and install the driver. I would remove driver from the constructor, and allow it to be null. Then in this function I would ASSERT(driver_ == nullptr);

**Comment by goaway** [comment]
Implemented above changes, but on hold pending related PR that supports an alternate approach.

**Comment by mattklein123** [comment]
@goaway should we just close this?

**Comment by goaway** [comment]
Sure, can always re-open if for some reason we decide we want it.

**Pull request closed without merging**